### PR TITLE
Fix ICP cutoff badness

### PR DESCRIPTION
**Pull request by matthewrmshin**
Reported via email by @benfitzpatrick. The new test would fail on current master.

### CODE REVIEW

**Comment by matthewrmshin** [comment]
It was broken by one of my recent optimisation refactor changes.

**Comment by matthewrmshin** [comment]
(Test failure related to pycodestyle, see #2465 fix.)

File: lib/cylc/config.py [hunk 1]
```diff
@@ -1562,7 +1562,7 @@ def generate_triggers(self, lexpression, left_nodes, right, seq,
                     ltaskdef.intercycle_offsets.add((None, seq))
                 else:
                     ltaskdef.intercycle_offsets.add(
-                        (str(first_point - last_point), seq))
+                        (str(-(last_point - first_point)), seq))
```
Comment (matthewrmshin) [inline lib/cylc/config.py]: Fix here. It is basically a revert of a recent change.

File: lib/cylc/taskdef.py [hunk 2]
```diff
@@ -163,9 +161,9 @@ def get_cleanup_cutoff_point(cls, my_point, offset_sequence_tuples):
...
-                # This is caused by future triggers - default to my_point.
-                return my_point
+            if max_cutoff_point < point:
+                # This is caused by future triggers - default to point.
+                return point
             return max_cutoff_point
-        # There aren't any dependent tasks in other cycles for my_point.
-        return my_point
+        # There aren't any dependent tasks in other cycles for point.
+        return point
```
Comment (matthewrmshin) [inline lib/cylc/taskdef.py]: I was originally looking at this method for clues of what went wrong, hence the minor refactor. There is actually nothing wrong with this method, apart from the rather awkward interface, which is simplified somewhat in this change.

**Comment by hjoliver** [comment]
Can you explain exactly what problem was caused by this bug?

**Comment by matthewrmshin** [comment]
Yes. `first_point - last_point` gave the nonsense answer `P-1DT18H`. `-(last_point - first_point)` gives the correct answer `-PT6H`. I guess this is ultimately a bug (or known badness?) in the *isodatetime* library.

**Comment by benfitzpatrick** [comment]
That does sound like a bad bug.

**Comment by benfitzpatrick** [comment]
See https://github.com/metomi/isodatetime/pull/82

**Comment by hjoliver** [comment]
(Travis CI fail is just due to pep8 -> pycodestyle)

**Pull request merged**